### PR TITLE
fix: Prevent ResizeManager from being clicked on safari, fix playerresize on firefox

### PR DESCRIPTION
**Pull request by BrandonOCasey**
Currently right clicking on the player on safari does not fire a `contextmenu` event, but it does on every other browser. Right clicking on the player in safari fires a `contextmenu` event on the `ResizeManager` iframe instead. The only change that needed to happen for right click to fire on the player again, was to hide `ResizeManager`.

EDIT:
I also found that the `playerresize` event does not work in Firefox. After removing `visibility: hidden;` it fires as we would expect, I added that change to this pull request, as I think it makes sense.

### CODE REVIEW

**Comment by BrandonOCasey** [comment]
Oh I see what you mean, when the video is actually playing it  gives a menu for the iframe, but before playback it gives a menu for the player. thats why I thought it was fixed. Looking further into this 👀

**Comment by BrandonOCasey** [comment]
Here is what I tried so far
1. `pointer-events: none` on the iframe
  * `contextmenu` is still hijacked
1. Wrap the iframe in a div and add `pointer-events: none` to it
  * `contextmenu` is still hijacked
1. Move the iframe before the video element
 * `contextmenu` is not hijacked but `playerresize` does not work
1. Try various `visiblitiy: none`/`display: none` on the iframe or a wrapper div
 * Causes `playerresize` not to fire but usually fixed the `contextmenu` issue

After all that I opted to re-trigger the `contextmenu` event on the player from the iframe, and everything appears to work as expected.

File: src/js/player.js [hunk 1]
```diff
@@ -4069,15 +4069,15 @@ Player.prototype.options_ = {
 
   // Included control sets
   children: [
+    'resizeManager',
```
Comment (gkatsev) [inline src/js/player.js]: maybe revert this change if we need to re-trigger the event?

File: src/js/resize-manager.js [hunk 2]
```diff
@@ -84,6 +86,18 @@ class ResizeManager extends Component {
...
+  /**
+   * Due to an issue on safari, the iframe for this object takes over
+   * right clicks on the player. So we have to re-trigger them from the
+   * iframe to the player.
+   * @see https://github.com/videojs/video.js/pull/5522
+   *
+   * @param {Event} e the contextmenu event that fired
+   */
+  handleContextmenu_(e) {
+    this.player_.trigger(e);
```
Comment (gkatsev) [inline src/js/resize-manager.js]: Should we preventDefault here? Thoughts @misteroneill?

File: src/js/player.js (hunk 1 above)
Comment (BrandonOCasey) [reply src/js/player.js]: OK, so I was going to remove this change and put resizeManager back at the bottom of the children list, but it breaks the controls for Safari. When I removed the `visibility: hidden;` css for the resize manager it made it so that the iframe was also intercepting other clicks on the video element. So we have to keep resizeManager at the top for controls to work correctly.

File: src/css/video-js.scss [hunk 3]
```diff
@@ -50,5 +50,5 @@
   width: 100%;
   height: 100%;
   border: none;
-  visibility: hidden;
```
Comment (BrandonOCasey) [inline src/css/video-js.scss]: Fixes firefox not firing `playerresize`

File: src/css/video-js.scss [hunk 4]
```diff
@@ -50,5 +50,5 @@
...
+  z-index: -1000;
```
Comment (BrandonOCasey) [inline src/css/video-js.scss]: Fixes safari and firefox iframe `contextmenu` hijacking

File: src/js/player.js (hunk 1 above)
Comment (gkatsev) [reply src/js/player.js]: If we still wanted to move it, I would put it after mediaLoader.

File: src/css/video-js.scss (hunk 4 above)
Comment (BrandonOCasey) [reply src/css/video-js.scss]: (firefox was started to hijack `contextmenu` once we removed `visibility: none`)
